### ISSUE TITLE
Incorrect hash values for large files (>=512MiB)

### ISSUE THREAD

**Issue by alexeyten** [digitalbazaar/forge#281]
For large files MD5 and SHA256 (and probably other) hashes calculate incorrect values.
There is bug in [MD5 `padBytes` calculation](https://github.com/digitalbazaar/forge/blob/master/js/md5.js#L137-142). E.g. if file size is exactly 512MiB, we get 0 as file size in padding.

The same bug in [SHA256 `padBytes` calculation](https://github.com/digitalbazaar/forge/blob/master/js/sha256.js#L143-148).

I suggest to get rid of `messageLength64` and just use `messageLength`. 56-bit lenth is enough for any reasonable file and `messageLength64` is filled incorrectly anyway.

My suggest for MD5 sum:

```diff
-    padBytes.putInt32Le(md.messageLength64[1] << 3);
-    padBytes.putInt32Le(
-      (md.messageLength64[0] << 3) | (md.messageLength64[0] >>> 28));
+    padBytes.putInt32Le(md.messageLength << 3);
+    padBytes.putInt32Le((md.messageLength / 0x20000000) >>> 0);
```

**Comment by dlongley** [comment]
Message digests have been rewritten completely for 0.7.x (see an example of the base class here: https://github.com/digitalbazaar/forge/blob/buffer-based-api/js/md.js). This includes fixes to message length bugs for long messages -- so that will be corrected once 0.7.x is out.

That being said, I'm going to keep this issue open until we can patch this bug in the 0.6.x version.

**Comment by r-shah** [comment]
I just found, this issue is already addressed. I have same issue while calculating SHA256 for file size >=512MB. I would appreciate if you could let me know if its fixes are coming anytime soon in 0.6.x version.

**Comment by r-shah** [comment]
@alexeyten 
Are you able to get around this issue? I tried changes suggested by you but does not work unfortunately. Please keep me posted, if you were able handle this issue somehow.

**Comment by dlongley** [comment]
This should be fixed now (version 0.6.37). Please reopen if it seems that the problem persists.

**Comment by r-shah** [comment]
Wonderful. Changes works like a charm.
